cartographer/mapping_2d/local_trajectory_builder.cc

```diff
@@ -148,38 +149,19 @@ LocalTrajectoryBuilder::AddAccumulatedRangeData(
       non_gravity_aligned_pose_prediction * gravity_alignment.inverse());
 
   transform::Rigid2d pose_estimate_2d;
-  ScanMatch(time, pose_prediction, gravity_aligned_range_data,
+  ScanMatch(time, pose_prediction, *gravity_aligned_range_data,
             &pose_estimate_2d);
   const transform::Rigid3d pose_estimate =
       transform::Embed3D(pose_estimate_2d) * gravity_alignment;
   extrapolator_->AddPose(time, pose_estimate);
 
-  last_pose_estimate_ = {
-      time, pose_estimate,
-      sensor::TransformPointCloud(
-          gravity_aligned_range_data.returns,
-          transform::Embed3D(pose_estimate_2d.cast<float>()))};
-
-  if (motion_filter_.IsSimilar(time, pose_estimate)) {
-    return nullptr;
-  }
-
-  // Querying the active submaps must be done here before calling
-  // InsertRangeData() since the queried values are valid for next insertion.
-  std::vector<std::shared_ptr<const Submap>> insertion_submaps;
-  for (const std::shared_ptr<Submap>& submap : active_submaps_.submaps()) {
-    insertion_submaps.push_back(submap);
-  }
-  active_submaps_.InsertRangeData(
-      TransformRangeData(gravity_aligned_range_data,
-                         transform::Embed3D(pose_estimate_2d.cast<float>())));
-
   sensor::AdaptiveVoxelFilter adaptive_voxel_filter(
```

Comment:
This seems unnecessary effort if the data is not inserted into the pose graph. Rather duplicate time/pose for performance and revert the change to the `InsertionResult`?